### PR TITLE
first pass at adding to month indexing time series

### PR DESCRIPTION
**Pull request by mgrover1**
I added a small bit to the [time series data](https://xarray.pydata.org/en/stable/user-guide/time-series.html) section of the user guide, detailing how to subset for a particular month from a dataset.

This solution is described in a [stack overflow solution](https://stackoverflow.com/questions/60791186/select-xarray-dataset-based-on-month), and figured users would benefit from knowing how to do this.

- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by dcherian** [comment]
Thanks @mgrover1 

I think this is clearer and potentially faster

``` python
ds.isel(time=ds.time.dt.month.isin([1]))
```

I noticed that this is not mentioned in the docs for [indexing using ``isin``](https://xarray.pydata.org/en/stable/user-guide/indexing.html#selecting-values-with-isin) so you could update that too if you're up for it.

**Comment by mgrover1** [comment]
> 
> I think this is clearer and potentially faster
> 
> ```python
> ds.isel(time=ds.time.dt.month.isin([1]))
> ```
> 
> I noticed that this is not mentioned in the docs for [indexing using `isin`](https://xarray.pydata.org/en/stable/user-guide/indexing.html#selecting-values-with-isin) so you could update that too if you're up for it.

I can add that example to the `isin` section... Within the example of looking at multiple months, the approach I included seemed more straight forward... what would subsetting for multiple months look like with using `isin`?

**Comment by dcherian** [comment]
oh oops

for a single month
```
ds.isel(ds.time.dt.month == 1)
```

for multiple months
```
ds.isel(ds.time.dt.month.isin([1, 2, 3]))
```

using groupby's new getitem method
```
ds.groupby("time.month")[1]
```
I think we want to encourage the first two; since groupby does a bunch of extra work.

I think we can add an indexing subsection under "Datetime components" showing the first two and link to that from the section you just edited.

**Comment by mgrover1** [comment]
Ohh.. neat! Yeah - I can add those few lines instead.

File: doc/user-guide/time-series.rst [hunk 1]
```diff
@@ -101,7 +101,7 @@ You can also select a particular time by indexing with a
 
     ds.sel(time=datetime.time(12))
 
-For more details, read the pandas documentation.
+For more details, read the pandas documentation and the section on indexing using the ``.dt`` acessor `datetime_component_indexing`_.
```
Comment (mathause) [inline doc/user-guide/time-series.rst:104]:
```suggestion
For more details, read the pandas documentation and the section on `Indexing Using Datetime Components <datetime_component_indexing>`_ (i.e. using the ``.dt`` acessor).
```

**Review by mathause — APPROVED** [review]
I have one minor suggestion but IMHO this is good to go.

**Comment by dcherian** [comment]
RTD failure is in plotting.rst which is unrelated. 

Thanks @mgrover1. I see this is your first contribution here. Welcome to xarray!

**Pull request merged**